### PR TITLE
ignore local peer ID when adding to a priority group

### PR DESCRIPTION
**Pull request by ordian**
In polkadot's validator discovery used also for validator-to-validator connections we may also accidentally include our own addresses when trying to change the priority group. This would result in [this warning](https://github.com/paritytech/polkadot/blob/f4897f74a69a6b3983c53b9bd8392556e0b47ce6/node/network/bridge/src/validator_discovery.rs#L282).

We could try to parse the mutiaddresses ourselves and filter out `service.local_peer_id()`, but a better default seems to be to ignore local peer id instead of failing the whole request.

### CODE REVIEW

**Review by bkchr — COMMENTED** [review]
Maybe a test?

File: client/network/src/service.rs [hunk 1]
```diff
@@ -1045,19 +1045,25 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 	/// invalid peer ID (which includes the local peer ID).
```
Comment (mxinden) [inline client/network/src/service.rs:1045]: This comment would need updating.

**Review by mxinden — COMMENTED** [review]
> we may also accidentally include our own addresses

Can you expand on how that is happening? Off the top of my head we filter out all addresses that [contain our peer id](https://github.com/paritytech/substrate/blob/master/client/authority-discovery/src/worker.rs#L533) in the authority discovery worker.

**Comment by ordian** [comment]
>  Off the top of my head we filter out all addresses that contain our peer id in the authority discovery worker.

Ah, thanks for pointing this out, I wasn't aware of this. In this case, that can't happen now.

**Comment by mxinden** [comment]
@ordian don't get me wrong, the above is my intuition. We might as well have a bug here. Have you seen `add_to_priority_group` return an error in the past, or did you propose this patch with an eye on the future?

**Comment by tomaka** [comment]
What we have noticed is that the peerset of a node is aware of itself, but we don't really know why this happens.

**Pull request closed without merging**